### PR TITLE
core(fr): enable traces artifact in timespan mode

### PR DESCRIPTION
**Pull request by patrickhulce**
**Summary**
Enables the traces artifact in timespan mode! 🎉  

- Enable the artifact itself.
- Limits many audits to `supportedMode: ['navigation']` if they don't immediately work in timespan mode (@adamraine to do a deeper pass on these for a breakup plan).
- Mark TBT notApplicable when throttlingMethod is simulate.
- Implement temporary alternate `wastedMs` computation method for opportunities to use until the rework.
- Replaces a few usages of `findMainDocument` with a new `findOptionalMainDocument` version that returns `undefined` instead of throwing.

See https://melodic-class.glitch.me/cerealshack-timespan.html for what a timespan report looks like after this.

**Related Issues/PRs**
ref #11313 
ref #12595

### CODE REVIEW

File: lighthouse-core/audits/byte-efficiency/byte-efficiency-audit.js [hunk 1]
```diff
@@ -122,7 +123,9 @@ class UnusedBytes extends Audit {
           LoadSimulator.request(simulatorOptions, context),
         ])
       )
-      .then(([result, graph, simulator]) => this.createAuditProduct(result, graph, simulator));
```
Comment (connorjclark) [inline lighthouse-core/audits/byte-efficiency/byte-efficiency-audit.js:125]: can you make this code use async/await for readability?

File: lighthouse-core/computed/metrics/timing-summary.js [hunk 2]
```diff
@@ -43,18 +43,18 @@ class TimingSummary {
     };
 
     const processedTrace = await ProcessedTrace.request(trace, context);
-    const processedNavigation = await ProcessedNavigation.request(processedTrace, context);
+    const processedNavigation = await requestOrUndefined(ProcessedNavigation, processedTrace);
     const speedline = await Speedline.request(trace, context);
-    const firstContentfulPaint = await FirstContentfulPaint.request(metricComputationData, context);
+    const firstContentfulPaint = await requestOrUndefined(FirstContentfulPaint, metricComputationData); // eslint-disable-line max-len
```
Comment (connorjclark) [inline lighthouse-core/computed/metrics/timing-summary.js:48]: when a good chunk of code is simple, repetitive and breaks max-len, I like to disable/enable the entire block of code.

File: lighthouse-core/computed/metrics/timing-summary.js [hunk 3]
```diff
@@ -89,27 +89,26 @@ class TimingSummary {
...
+      observedFirstContentfulPaintAllFramesTs: processedNavigation && processedNavigation.timestamps.firstContentfulPaintAllFrames, // eslint-disable-line max-len
+      observedFirstMeaningfulPaint: processedNavigation && processedNavigation.timings.firstMeaningfulPaint, // eslint-disable-line max-len
+      observedFirstMeaningfulPaintTs: processedNavigation && processedNavigation.timestamps.firstMeaningfulPaint, // eslint-disable-line max-len
+      observedLargestContentfulPaint: processedNavigation && processedNavigation.timings.largestContentfulPaint, // eslint-disable-line max-len
+      observedLargestContentfulPaintTs: processedNavigation && processedNavigation.timestamps.largestContentfulPaint, // eslint-disable-line max-len
+      observedLargestContentfulPaintAllFrames: processedNavigation && processedNavigation.timings.largestContentfulPaintAllFrames, // eslint-disable-line max-len
+      observedLargestContentfulPaintAllFramesTs: processedNavigation && processedNavigation.timestamps.largestContentfulPaintAllFrames, // eslint-disable-line max-len
```
Comment (connorjclark) [inline lighthouse-core/computed/metrics/timing-summary.js:105]: when can we do `a?.b`

**Review by connorjclark — APPROVED** [review]

File: lighthouse-core/computed/metrics/timing-summary.js (hunk 3 above)
Comment (patrickhulce) [reply lighthouse-core/computed/metrics/timing-summary.js:105]: not soon enough! haha

**Pull request merged**